src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -122,48 +122,34 @@ public function collect(Request $request, Response $response, \Exception $except
         }
 
         if (isset($this->controllers[$request])) {
-            $controller = $this->controllers[$request];
-            if (is_array($controller)) {
-                try {
-                    $r = new \ReflectionMethod($controller[0], $controller[1]);
-                    $this->data['controller'] = array(
-                        'class' => is_object($controller[0]) ? get_class($controller[0]) : $controller[0],
-                        'method' => $controller[1],
-                        'file' => $r->getFileName(),
-                        'line' => $r->getStartLine(),
-                    );
-                } catch (\ReflectionException $e) {
-                    if (is_callable($controller)) {
-                        // using __call or  __callStatic
-                        $this->data['controller'] = array(
-                            'class' => is_object($controller[0]) ? get_class($controller[0]) : $controller[0],
-                            'method' => $controller[1],
-                            'file' => 'n/a',
-                            'line' => 'n/a',
-                        );
-                    }
-                }
-            } elseif ($controller instanceof \Closure) {
-                $r = new \ReflectionFunction($controller);
-                $this->data['controller'] = array(
-                    'class' => $r->getName(),
-                    'method' => null,
-                    'file' => $r->getFileName(),
-                    'line' => $r->getStartLine(),
-                );
-            } elseif (is_object($controller)) {
-                $r = new \ReflectionClass($controller);
-                $this->data['controller'] = array(
-                    'class' => $r->getName(),
-                    'method' => null,
-                    'file' => $r->getFileName(),
-                    'line' => $r->getStartLine(),
-                );
-            } else {
-                $this->data['controller'] = (string) $controller ?: 'n/a';
-            }
+            $this->data['controller'] = $this->parseController($this->controllers[$request]);
             unset($this->controllers[$request]);
         }
+
+        if ($parentRequestAttributes = $request->attributes->get('_forwarded')) {
+            if ($parentRequestAttributes instanceof ParameterBag) {
+                $parentRequestAttributes->set('_forward_token', $response->headers->get('x-debug-token'));
+            }
+        }
+
+        if ($request->attributes->has('_forward_controller')) {
+            $this->data['forward'] = array(
+                'token' => $request->attributes->get('_forward_token'),
+                'controller' => $request->attributes->get('_forward_controller'),
+            );
+        }
+
+        if ($request->hasSession() && $request->getSession()->has('sf_redirect')) {
+            $this->data['redirect'] = $request->getSession()->get('sf_redirect');
+            $request->getSession()->remove('sf_redirect');
```

Comment:
In case of chained redirections $response, each request collects its own previous $request profile, it will then also appear in wdt while on profiler intercept redirection page.

![chained_redirect](https://cloud.githubusercontent.com/assets/10107633/12719686/060dabb2-c8f7-11e5-8f1f-12cdd08f9fa0.jpg)